src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!timer_stop_flag_) {
```

Comment:
@BrandonHe I can see that onTimer function will be called once, and then it will be just looped in condition variable while it won't be stopped in destructor. I can't understand why do you use timer for this purpose, you should use thread logic for this purpose. You just implemented timer inside timer callback, it is confusing. 

Also within you realisation "expired logic" will be executed each time new request is added (because of NotifyOne in UpdateTimeout), it is redundant.